src/Connector.php

```diff
@@ -81,17 +81,17 @@ public function create(
     private function listenForPackets(Stream $stream)
     {
         $stream->on('data', function ($rawData) use ($stream) {
-            $messages = $this->splitMessage($rawData);
-            foreach ($messages as $data) {
-                try {
-                    $message = Factory::getByMessage($this->version, $data);
-                    $stream->emit($message::EVENT, [$message]);
-
-                    echo "received:\t" . get_class($message) . "\n";
-                } catch (\InvalidArgumentException $ex) {
-
+            try {
+                foreach (Factory::getNextPacket($this->version, $rawData) as $packet) {
+                    $stream->emit($packet::EVENT, [$packet]);
+                    echo "received:\t" . get_class($packet) . PHP_EOL;
                 }
             }
+            //TODO InvalidPacketException or something
+            catch (\InvalidArgumentException $e) {
+                //TODO Actually, the spec says to disconnect if you receive invalid data.
```

Comment:
I think that's what this says.

> Unless stated otherwise, if either the Server or Client encounters a protocol violation, it MUST close the Network Connection on which it received that Control Packet which caused the protocol violation [MQTT-4.8.0-1]

http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Ref381955543